packages/astro/src/core/server-islands/vite-plugin-server-islands.ts

```diff
@@ -37,6 +37,20 @@ export function vitePluginServerIslands({ settings }: { settings: AstroSettings
 					if (astro?.serverComponents.length) {
 						for (const comp of astro.serverComponents) {
 							if (!settings.serverIslandNameMap.has(comp.resolvedPath)) {
+								if (!settings.adapter) {
+									logger.error(
+										'islands',
+										"You tried to use a server island without an adapter. This an error and your project won't build.",
+									);
+								}
+
+								if (settings.buildOutput !== 'server') {
+									logger.error(
+										'islands',
+										'You tried to use a server island, but your output isn\'t `"server"`. Use an adapter that support server output.',
```

Comment:
Since you don't *need* `output: 'server'` for server islands, and since changing the output value is a pretty large change to someone's project, I would hesitate to give a message that says this is required.

Is there a situation where someone really would need a) THIS message, forcing them to use `output: 'server'` and b) the previous message wouldn't work?